### PR TITLE
feat: add a namespace gc command

### PR DESCRIPTION
**Pull request by josegonzalez**
This command will cleanup empty namespaces automatically and also supports a dry-run mode.

### CODE REVIEW

File: command/namespace/gc.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	nomadClient, err := api.NewClient(api.DefaultConfig())
+	if err != nil {
+		return err
+	}
+
+	var deletableNamespaces []*api.Namespace
+	namespaces, _, err := nomadClient.Namespaces().List(nil)
+	for _, namespace := range namespaces {
+
+		jobs, _, err := nomadClient.Jobs().List(&api.QueryOptions{Namespace: namespace.Name})
```
Comment (ascendantlogic) [inline command/namespace/gc.go:21]: Reading the docs for the `List` function I didn't see anything regarding controlling if this returns only running jobs or not. Will this skip namespaces that have dead jobs in them?

Comment (josegonzalez) [reply command/namespace/gc.go:21]: Good question. When you say dead, do you mean stopped jobs or do you mean jobs that nomad has marked as dead?

Comment (ascendantlogic) [reply command/namespace/gc.go:21]: Yeah, is this only going to return running jobs or will it return all service jobs, running, dead or pending, within the specified namespace?

Comment (josegonzalez) [reply command/namespace/gc.go:21]: It shows all jobs, regardless of status, within a namespace. Just checked in our staging environment.

Comment (ascendantlogic) [reply command/namespace/gc.go:21]: Ok so then do we want to skip Namespace GC if there's dead jobs or just if there's running/pending?

Comment (ascendantlogic) [reply command/namespace/gc.go:21]: If there's just dead jobs in a Namespace I'd argue it should be a candidate for cleanup. How often will this run? Once daily?

Comment (josegonzalez) [reply command/namespace/gc.go:21]: We want to skip if there are _any_ jobs. Potentially we could have another flag that is like `ignore-dead` or something (not sure on naming) that would not count dead jobs in the check for an empty namespace.

We need to delete jobs first when deleting a non-empty namespace, right?

**Review by ascendantlogic — APPROVED** [review]

**Pull request merged**